### PR TITLE
docs: update custom extractor tutorial

### PR DESCRIPTION
**Pull request by jared-rimmer**
I have ported over and merged content from:

https://meltano.com/blog/how-to-build-a-custom-extractor-with-meltano/

with the existing custom extractor tutorial in an attempt to making it a one stop shop for people wanting to get started with a creating a custom extractor.

Open questions

- **Terminology**: I've used "custom extractor" throughout and changed references from tap, plugin, project, source. What are your thoughts on this?
- **SEO**: As a lot of the content is a literal copy and paste from the blog it might have implications of duplicate content. Would you prefer a more thorough rewrite?

### CODE REVIEW

**Comment by tayloramurphy** [comment]
@afolson you know more about SEO than me - what's our preference on this? cc @lauraperezco

**Comment by lauraperezco** [comment]
> @afolson you know more about SEO than me - what's our preference on this? cc @lauraperezco

@tayloramurphy same as the ELT page applies here, the main website has preference over subpages for SEO, so Jared's assesment is _correct_! I think ideally we should replace the current content in docs with a short introduction that takes you to the blog link for the full tutorial (and maybe reference some other blog-tutorials in the future to make this section in docs more complete while giving more authority to our website)

File: CHANGELOG.md [hunk 1]
```diff
@@ -106,6 +106,7 @@ This project adheres to [Semantic Versioning](http://semver.org/) and [Keep a Ch
 ---
 
 ### New
+* Update custom extractor tutorial content
```
Comment (edgarrmondragon) [inline CHANGELOG.md:109]: This isn't needed as the changelog entry will be taken from the merge commit from this PR 🙂 
```suggestion
```

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]

**Comment by jared-rimmer** [comment]
@lauraperezco happy to update with your suggestions 👍 

From the [Slack thread](https://meltano.slack.com/archives/C013Z450LCD/p1661512073303649) there was an aim to make the docs the place where users can find all this type of information. Linking out to blogs still achieves this

**Comment by sbalnojan** [comment]
Hey @jared-rimmer, no the PR can stay as it is, everything in once place. 

But FWIW we're aware that we need to figure out a guide for what goes where, that's still WIP ;)

File: docs/src/_tutorials/custom-extractor.md [hunk 2]
```diff
@@ -2,39 +2,392 @@
...
+
+Run the following command to create the project files from the cookiecutter template:
+
+```
+cookiecutter https://github.com/meltano/sdk --directory="cookiecutter/tap-template"
+```
+
+After running the above command, you will be prompted to configure your project.
+
+- Choose jsonplaceholder as your source name
```
Comment (edgarrmondragon) [inline docs/src/_tutorials/custom-extractor.md:78]:
```suggestion
- Type `jsonplaceholder` as your source name
```

File: docs/src/_tutorials/custom-extractor.md [hunk 3]
```diff
@@ -2,39 +2,392 @@
...
+
+3 - Basic Auth
+
+4 - OAuth2
+
+5 - JWT
+
+6 - Custom or N/A
+
+Choose from 1, 2, 3, 4, 5, 6 [1]: 6
```
Comment (edgarrmondragon) [inline docs/src/_tutorials/custom-extractor.md:119]:
```suggestion
1 - API Key
2 - Bearer Token
3 - Basic Auth
4 - OAuth2
5 - JWT
6 - Custom or N/A
Choose from 1, 2, 3, 4, 5, 6 [1]: 6
```

File: docs/src/_tutorials/custom-extractor.md [hunk 4]
```diff
@@ -2,39 +2,392 @@
...
+
+Select include_cicd_sample_template:
+
+1 - GitHub
+
+2 - None (Skip)
+
+Choose from 1, 2 [1]:
```
Comment (edgarrmondragon) [inline docs/src/_tutorials/custom-extractor.md:127]:
```suggestion
1 - GitHub
2 - None (Skip)
Choose from 1, 2 [1]:
```

File: docs/src/_tutorials/custom-extractor.md [hunk 5]
```diff
@@ -2,39 +2,392 @@
...
+```
+
+The result of the above command is a new directory—tap-jsonplaceholder—that contains boilerplate code for developing your tap and also a meltano.yml file that you can use to test your custom extractor.
```
Comment (edgarrmondragon) [inline docs/src/_tutorials/custom-extractor.md:130]:
```suggestion
The result of the above command is a new directory `tap-jsonplaceholder` that contains boilerplate code for developing your tap and also a `meltano.yml` file that you can use to test your custom extractor.
```

File: docs/src/_tutorials/custom-extractor.md [hunk 6]
```diff
@@ -2,39 +2,392 @@
...
+
+th.Property("id", th.IntegerType),
+
+th.Property("name", th.StringType),
+
+th.Property("email", th.StringType),
+
+th.Property("body", th.StringType),
+
+).to_dict()
```
Comment (edgarrmondragon) [inline docs/src/_tutorials/custom-extractor.md:230]:
```suggestion
class CommentsStream(jsonplaceholderStream):
    primary_keys = ["id"]
    path = '/comments'
    name = "comments"
    schema = th.PropertiesList(
        th.Property("postId", th.IntegerType),
        th.Property("id", th.IntegerType),
        th.Property("name", th.StringType),
        th.Property("email", th.StringType),
        th.Property("body", th.StringType),
    ).to_dict()
```

**Review by edgarrmondragon — DISMISSED** [review]

**Review by sbalnojan — APPROVED** [review]
all changes adressed!

**Pull request merged**